palindrome_checker.cpp

```diff
@@ -1,6 +1,43 @@
+/*
+______________ Assumptions _____________
+1. LinkedList contains int or char
+2. LinkedList size >= 1
+3. Function throws invalid input Exception if input is 
+	1. NULL
+	2. ListNode elements is not of type int or char
+4. Function Returns bool 
+
+
+
+
+
+____________ Examples ________________
```

Comment:
Good test cases - one extra one to include for palindromes is a simple list with an even number of elements.